core/app/services/spree/checkout/add_store_credit.rb

```diff
@@ -0,0 +1,49 @@
+module Spree
+  module Checkout
+    class AddStoreCredit
+      prepend Spree::ServiceModule::Base
+
+      def call(order:, amount: nil)
+        @order = order
+        return failed unless @order
+
+        @order.payments.store_credits.where(state: :checkout).map(&:invalidate!)
```

Comment:
@caudatecoder probably it would be a good idea to wrap both of those operations (invalidation + applying credit) in a single DB transaction, what do you think?

Reply:
@damianlegawiec makes sense to me
Should be fine 👍